src/MainWindow.vala

```diff
@@ -463,9 +463,8 @@ namespace Terminal {
                             return true;
                         } else if (!current_terminal.has_foreground_process ()) {
                             /* Ignore returns being sent to a foreground process */
-                            current_terminal.remember_position ();
-                            get_simple_action (ACTION_SCROLL_TO_LAST_COMMAND).set_enabled (true);
                             current_terminal.remember_command_end_position ();
```

Comment:
Remember cursor_command_start_ position when pressing enter.

```vala
Timeout.add (100, () => {
    current_terminal.remember_command_start_position ();
    return false;
})
```

Reply:
Logic breaks down if command needs more then 100ms seconds to return. `remembered_command_start_row` should be remembered as soon as command returns.

Reply:
Nevermind I found a better solution see comment below.